crates/sui-types/src/gas.rs

```diff
@@ -97,12 +97,36 @@ macro_rules! ok_or_gas_balance_error {
     };
 }
 
+/// Summary of the charges in a transaction.
+/// Storage is charged independently of computation.
+/// There are 3 parts to the storage charges:
+/// `storage_cost`: it is the charge of storage at the time the transaction is executed.
+///                 The cost of storage is the number of bytes of the objects being mutated
+///                 multiplied by a variable storage cost per byte
+/// `storage_rebate`: this is the amount a user gets back when manipulating an object.
+///                   The `storage_rebate` is the `storage_cost` for an object minus fees.
+/// `non_refundable_storage_fee`: not all the value of the object storage cost is
+///                               given back to user and there is a small fraction that
+///                               is kept by the system. This value tracks that charge.
+///
+/// When looking at a gas cost summary the amount charged to the user is
+/// `computation_cost + storage_cost - storage_rebate`
+/// and that is the amount that is deducted from the gas coins.
+/// `non_refundable_storage_fee` is collected from the objects being mutated/deleted
+/// and it is tracked by the system in storage funds.
 #[derive(Eq, PartialEq, Clone, Debug, Default, Serialize, Deserialize, JsonSchema)]
 #[serde(rename_all = "camelCase")]
 pub struct GasCostSummary {
+    /// Cost of computation/execution
     pub computation_cost: u64,
+    /// Storage cost, it's the sum of all storage cost for all objects created or mutated.
     pub storage_cost: u64,
+    /// The amount of storage cost refunded to the user for all object manipulated in the
+    /// transaction.
     pub storage_rebate: u64,
+    /// The fee for the rebate. At any point in time for a given transaction
+    /// storage_cost = storage_rebate + non_refundable_storage_fee.
```

Comment:
This comment does not look right...

Reply:
Storage cost is the amount that we are charging a user for any objects created. Created objects include newly minted objects + any new versions of objects that were mutated.

This storage cost should and needs to be independent of that actual storage cost charged for a transaction.

Any objects deleted, including the older versions of objects mutated, have the storage field on the objects added up to a pool of "potential rebate". This rebate then is reduced by the "nonrefundable rate" such that:

potential rebate (storage cost of deleted objects) = storage_rebate + non_refundable_storage_fee

Reply:
> Storage cost is the amount that we are charging a user for any objects created. Created objects include newly minted objects + any new versions of objects that were mutated.

agreed, mutated objects are sort of deleted and recreated, so to speak

> This storage cost should and needs to be independent of that actual storage cost charged for a transaction.

not sure what you mean here, mind to articulate more?

> Any objects deleted, including the older versions of objects mutated, have the storage field on the objects added up to a pool of "potential rebate". This rebate then is reduced by the "nonrefundable rate" such that:
potential rebate (storage cost of deleted objects) = storage_rebate + non_refundable_storage_fee

fair way to describe it

Reply:
>>This storage cost should and needs to be independent of that actual storage cost charged for a transaction.

>not sure what you mean here, mind to articulate more?

Deleted that line...idk what i was trying to say, maybe that the rebate needs to be independent of the storage cost charged?